src/aggregator/integration/client.go

```diff
@@ -85,7 +86,12 @@ func (c *client) writePassthroughMetricWithMetadata(
 }
 
 func (c *client) flush() error {
-	return c.aggClient.Flush()
+	err := c.aggClient.Flush()
+	if err == nil || strings.Contains(err.Error(), aggclient.ErrFlushInProgress.Error()) {
```

Comment:
I originally wanted to use errors.Is() but couldn't since Flush() returns a multierr-converted-to-a-single-concatenated error using FinalError() which simply appends all errors thus not conforming to the format expected by errors.Is()

Reply:
Fair, that is annoying haha. Ultimately we need to replace our `x/errors` package with `multierr` and Go's builtin error wrapping, but that's a project for another day haha. 

Looks like someone wrote the right helper for this actually:

```
// Contains returns true if any of the errors match the provided error using the Is check.
func (e MultiError) Contains(err error) bool {
	if errors.Is(e.err, err) {
		return true
	}
	for _, e := range e.errors {
		if errors.Is(e, err) {
			return true
		}
	}
	return false
}
```

; should probably use that?

Secondary question though: say we have two errors, one `ErrFlushInProgress` and one other (nonexpected) error. Shouldn't we ignore the ErrFlushInProgress but return the nonexpected error?